src/main/scala/firrtl/Compiler.scala

```diff
@@ -49,13 +49,38 @@ case class CircuitState(
   annotations: Option[AnnotationMap] = None,
   renames: Option[RenameMap] = None)
 
-/** Current form of the Firrtl Circuit */
+/** Current form of the Firrtl Circuit
+  *
+  * Form is a measure of addition restrictions on the legality of a Firrtl
+  * circuit.  There is a notion of "highness" and "lowness" implemented in the
+  * compiler by extending [[scala.math.Ordered]]. "Lower" forms add additional
+  * restrictions compared to "higher" forms.
+  *
+  * Thus, [[ChirrtlForm]] has the fewest requirements while [[LowForm]] has the most.
+  */
 sealed abstract class CircuitForm(private val value: Int) extends Ordered[CircuitForm] {
   def compare(that: CircuitForm): Int = this.value - that.value
 }
+/** Chirrtl Form
+  *
+  * The form of the circuit emitted by Chisel. Not a true Firrtl form.
+  * @todo Detailed description
+  */
 final case object ChirrtlForm extends CircuitForm(3)
+/** High Form
+  *
+  * @todo Detailed description
+  */
 final case object HighForm extends CircuitForm(2)
+/** Middle Form
+  *
+  * @todo Detailed description
```

Comment:
All widths have been inferred, when's have been expanded, and all connections have been expanded.